File: algorithms/0014-longest-common-prefix.go
func longestCommonPrefix(strs []string) string {
	if len(strs) == 0 {
		return ""
	}
	length := len(strs[0])
	for i := 1; i < len(strs); i++ {
		var compare_len int
		if length < len(strs[i]) {
			compare_len = length
		} else {
			compare_len = len(strs[i])
		}
		found := false
		for j := 0; j < compare_len; j++ {
			if strs[0][j] != strs[i][j] {
				length = j
				found = true
				break
			}
		}
		if !found {
			length = compare_len
		}
	}
	return strs[0][:length]
}

File: algorithms/0014-longest-common-prefix.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        if not strs:
            return ""
        length = len(strs[0])
        for each in strs[1:]:
            compare_len = min(length, len(each)
            for index in range(compare_len)):
                if strs[0][index] != each[index]:
                    length = index
                    break
            else:
                length = compare_len
        return strs[0][:length]
